src/jwplayer/players/DesktopArticleVideoPlayer/DesktopArticleVideoPlayer.tsx

```diff
@@ -50,23 +57,35 @@ export const DesktopArticleVideoPlayerContent: React.FC<DesktopArticleVideoPlaye
 							isScrollPlayer ? styles.desktopArticleVideoWrapperScrollPlayer : styles.desktopArticleVideoWrapper
 						}
 					>
-						{adComplete && (
+						{adEngineComplete && jwpAdsSetupComplete && (
 							<div>
 								<div className={styles.topBar}>
 									{!isScrollPlayer && <UnmuteButton />}
 									{isScrollPlayer && (
 										<CloseButton dismiss={() => setDismissed(true)} iconColor={'#fff'} className={styles.closeButton} />
 									)}
 								</div>
-								<JwPlayerWrapper
-									getDismissed={getDismissed}
-									config={getArticleVideoConfig(videoDetails)}
-									onReady={(playerInstance) => {
-										articlePlayerOnReady(videoDetails, playerInstance);
-										setIsPlayerReady(true);
-									}}
-									stopAutoAdvanceOnExitViewport={false}
-								/>
+								{strategyRulesEnabled ? (
+									<JwPlayerWrapperWithStrategyRules
+										getDismissed={getDismissed}
+										config={getArticleVideoConfig(videoDetails)}
+										onReady={(playerInstance) => {
+											articlePlayerOnReady(videoDetails, playerInstance);
+											setIsPlayerReady(true);
+										}}
+										stopAutoAdvanceOnExitViewport={false}
+									/>
+								) : (
+									<JwPlayerWrapper
+										getDismissed={getDismissed}
+										config={getArticleVideoConfig(videoDetails)}
+										onReady={(playerInstance) => {
+											articlePlayerOnReady(videoDetails, playerInstance);
+											setIsPlayerReady(true);
+										}}
+										stopAutoAdvanceOnExitViewport={false}
+									/>
+								)}
```

Comment:
I'm not very familiar with React, but in Vue, we have dynamic components that simplify templates and reduce code duplication. Is there a way to achieve something similar in React? I did a quick search and found some common approaches like conditional rendering, using component maps, and utilizing functions or render props. In essence, is it feasible to reduce code duplication, or is the current implementation a temporary solution?

Reply:
My hope is that it's just a temporary and after we conduct the experiment we'll remove `JwPlayerWrapper` component 😈 But if for some reason we'll learn that we need both options it's worth looking more into the solutions you've found 👌

Reply:
Sure, that sounds good. Thanks for the clarification.